driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy

```diff
@@ -981,4 +982,40 @@ class AggregatesFunctionalSpecification extends OperationFunctionalSpecification
         then:
         results == [Document.parse('{b: 1, _id: 7}')]
     }
+
+    @IgnoreIf({ !serverVersionAtLeast(4, 3) })
+    def '$unionWith'() {
+        given:
+        def coll1Helper = getCollectionHelper(new MongoNamespace(getDatabaseName(), 'coll1'))
+        def coll2Helper = getCollectionHelper(new MongoNamespace(getDatabaseName(), 'coll2'))
+
+        coll1Helper.drop()
+        coll2Helper.drop()
+
+        coll1Helper.insertDocuments(
+                Document.parse('{ "name1" : "almonds" }'),
+                Document.parse('{ "name1" : "cookies" }'))
+
+        coll2Helper.insertDocuments(
+                Document.parse('{ "name2" : "cookies" }'),
+                Document.parse('{ "name2" : "cookies" }'),
+                Document.parse('{ "name2" : "pecans" }'))
+
+        def pipeline = asList(match(eq('name2', 'cookies')), project(fields(excludeId(), computed('name', '$name2'))))
```

Comment:
I don't think there are any ordering guarantees for this stage, so to be safe against future behavioral changes, add a sort by name as the last stage so that the assertion on line 1011 will always succeed.

Reply:
I have added `sort` as the last stage of the pipeline.